### PR TITLE
MUSARK-558

### PR DESCRIPTION
**Pull request by jarlah**
Fix actor suggest component hammering the service for every typed character.

Fix actor not displaying on observation and control page after refresh (F5)

### CODE REVIEW

File: src/state/RxState.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return class Connect extends Component {
+      state$ = providedState$;
+
+      static contextTypes = {
+        state$: PropTypes.object // .isRequired
+      };
+
+      constructor(props, context) {
+        super(props, context);
+        this.state$ = this.context.state$ || this.state$;
```
Comment (jarlah) [inline src/state/RxState.js:29]: Maybe i can swap this OR condition. So that some connected components use their own state

File: package.json [hunk 2]
```diff
@@ -51,7 +51,7 @@
     "redux-form": "6.1.1",
     "redux-logger": "2.7.0",
     "reselect": "^2.5.4",
-    "rx": "^4.1.0",
+    "rxjs": "^5.0.0-rc.1",
```
Comment (kpmeen) [inline package.json]: Dynamic version ranges should be avoided.

File: src/state/RxState.js (hunk 1 above)
Comment (kpmeen) [reply src/state/RxState.js:29]: Make a note of it in the form of a comment or something. Not necessary to spend time on that now. But I do agree. Taking this pattern into the larger codebase that would be preferable.

**Review by kpmeen — COMMENTED** [review]
Looks good to me...The only thing I could see was the version range on the RxJS dependency

**Review by kpmeen — APPROVED** [review]

**Pull request merged**